app/src/androidTest/java/co/nimblehq/compose/crypto/ui/screen/detail/DetailScreenTest.kt

```diff
@@ -0,0 +1,170 @@
+package co.nimblehq.compose.crypto.ui.screen.detail
+
+import androidx.activity.compose.setContent
+import androidx.compose.ui.test.*
+import androidx.compose.ui.test.junit4.createAndroidComposeRule
+import co.nimblehq.compose.crypto.R
+import co.nimblehq.compose.crypto.domain.usecase.GetCoinDetailUseCase
+import co.nimblehq.compose.crypto.domain.usecase.GetCoinPricesUseCase
+import co.nimblehq.compose.crypto.extension.toFormattedString
+import co.nimblehq.compose.crypto.test.MockUtil.coinDetail
+import co.nimblehq.compose.crypto.test.MockUtil.coinPrices
+import co.nimblehq.compose.crypto.ui.BaseScreenTest
+import co.nimblehq.compose.crypto.ui.components.chartintervals.TimeIntervals
+import co.nimblehq.compose.crypto.ui.navigation.AppDestination
+import co.nimblehq.compose.crypto.ui.screens.MainActivity
+import co.nimblehq.compose.crypto.ui.screens.detail.*
+import co.nimblehq.compose.crypto.ui.screens.home.FIAT_CURRENCY
+import io.mockk.every
+import io.mockk.mockk
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.flowOf
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import kotlin.math.abs
+
+@ExperimentalCoroutinesApi
+class DetailScreenTest : BaseScreenTest() {
+
+    @get:Rule
+    val composeAndroidTestRule = createAndroidComposeRule<MainActivity>()
+
+    private val errorGeneric: String
+        get() = composeAndroidTestRule.activity.getString(R.string.error_generic)
+
+    private val mockGetCoinDetailUseCase: GetCoinDetailUseCase = mockk()
+    private val mockGetCoinPricesUseCase: GetCoinPricesUseCase = mockk()
+
+    private lateinit var detailViewModel: DetailViewModel
+
+    private var appDestination: AppDestination? = null
+
+    @Before
+    fun setUp() {
+        composeAndroidTestRule.activity.setContent {
+            DetailScreen(
+                coinId = "",
+                viewModel = detailViewModel,
+                navigator = { destination ->
+                    appDestination = destination
+                }
+            )
+        }
+    }
+
+    @Test
+    fun when_navigate_to_detail_screen_it_show_loading() {
+        composeAndroidTestRule.activity.setContent {
+            DetailScreen(
+                coinId = "",
+                navigator = { destination ->
+                    appDestination = destination
+                }
+            )
+        }
+
+        initDetailViewModel()
```

Comment:
@kaungkhantsoe Just curious, why do we not init the ViewModel in `setUp`?

Reply:
@doannimble Sometime we have to init the `ViewModel` after `every { } returns flow { }` or we get `no answer found for: Repo` error because the ViewModel passed to the screen Composable doesn't have mocked answer. But here, we can move to `setup`. I will update.